### DISCUSSION TITLE
SAME_UPPER and SAME_LOWER clarification

### DISCUSSION THREAD

**Discussion by kckeiks** [onnx/onnx#6200] [Q&A] [answered]
Hello,

The CONV section in the spec says regarding auto_pad=SAME_UPPER/SAME_LOWER:

> The padding is split between the two sides equally or almost equally (depending on whether it is even or odd). In case the padding is an odd number, the extra padding is added at the end for SAME_UPPER and at the beginning for SAME_LOWER.

What does it mean by "split between two sides"? Sides of what? What is it referring to that has a "beginning" and "end"? 

Is this the same idea as tensorflow's left/right and top/bottom padding? 

An example would be helpful. Thanks in advance!

**Comment by justinchuby** [comment]
Cc @gramalingam

**Comment by gramalingam** [comment, marked as answer]
Let's restrict attention to a single-dimension. Given an input `[1, 2, 3, 4]`, padding with a zero at the beginning means `[0, 1, 2, 3, 4]`, padding at the end means `[1, 2, 3, 4, 0]`. Yes, I would say left/right or top/bottom means the same thing (in the context of rows and columns for a 2-dimensional matrix).
